src/Files.App/UserControls/AddressToolbar.xaml.cs

```diff
@@ -15,8 +15,17 @@ namespace Files.App.UserControls
 	public sealed partial class AddressToolbar : UserControl
 	{
 		private readonly IUserSettingsService userSettingsService = Ioc.Default.GetRequiredService<IUserSettingsService>();
+		private readonly MainPageViewModel MainPageViewModel = Ioc.Default.GetRequiredService<MainPageViewModel>();
```

Comment:
Accessing `MainPageViewModel` from AddressToolbar isn't the best practice, but it'll do for now given the state of the rest of the control.

Reply:
I heisted to do this and I wanted to refer to `AddressToolbarViewModel` from `MainPage` using adaptive window trigger in its XAML and using DI'd instance of `AddressToolbarViewModel`.
But as of now, the view model instance differs across tab item and saved in each associated tab in `ShellPanesPage.ActivePaneOrColumn.AddressToolbarViewModel`, and `ShellPanesPage` exists as `PaneHolder` public property in `SidebarViewModel`:

SidebarViewModel.PaneHolder.ActivePaneOrColumn.AddressToolbarViewModel.

I'm not sure why on earth `SidebarViewModel` has that instance in the first place, but in my plan of refactoring sidebar I'll remove SidebarViewModel and move all of it into MainPageViewModel because in order to standardize the usage also for the properties window and the settings dialog. And then, I'll make AddressToolbarViewModel singleton instance so we can instance it via DI from everywhere.

---

While this doesn't fix the underlying issue and I don't like it (because it's the same thing as referencing to SidebarViewModel), we can refer to ContentPageContext.ShellPage.ActivePaneOrColumn.AddressToolbarViewModel in MainPage XAML.